src/libs/actions/Task.ts

```diff
@@ -1056,7 +1056,13 @@ function deleteTask(report: OnyxEntry<OnyxTypes.Report>) {
     Report.notifyNewAction(report.reportID, currentUserAccountID);
 
     if (shouldDeleteTaskReport) {
-        return ROUTES.REPORT_WITH_ID.getRoute(parentReport?.reportID ?? '-1');
+        if (parentReport?.reportID) {
+            return ROUTES.REPORT_WITH_ID.getRoute(parentReport.reportID);
+        }
+        const mostRecentReportID = Report.getMostRecentReportID(report);
+        if (mostRecentReportID) {
+            return ROUTES.REPORT_WITH_ID.getRoute(mostRecentReportID);
+        }
```

Comment:
@MonilBhavsar the deleteTask function now needs to return the route and the navigation will happen after the delete confirmation modal hides,
https://github.com/Expensify/App/blob/98124799a639606573877bf6eb38554ee2939a22/src/pages/ReportDetailsPage.tsx#L783-L788

so I create `getMostRecentReportID` to return the most recent report instead of using `navigateToMostRecentReport`.

Reply:
Okay thanks for clarifying.
Any idea, in which case this is falsy because we fallback to concierge chat which should always exist.

if (mostRecentReportID) {
    return ROUTES.REPORT_WITH_ID.getRoute(mostRecentReportID);
}

Reply:
`conciergeChatReportID` has a type of `boolean | undefined`. 1 case it will be undefined that I can think of is when the app is still loading for the first time, but at this point, the user shouldn't have access to the task report.

But even with that information, the type could still be undefined, and `.getRoute` typing doesn't accept undefined reportID. So, the choice is:
```
if (mostRecentReportID) { 
    return ROUTES.REPORT_WITH_ID.getRoute(mostRecentReportID);
}
```

OR

```
return ROUTES.REPORT_WITH_ID.getRoute(mostRecentReportID ?? '-1');
```

(-1 because that's the agreed fallback value)

Reply:
The latter looks good and explains the case well
And in former, we're not falling back to -1 at all

Reply:
But if we fall back to -1, then the not found page will show again (/r/-1). In the worst scenario, if the concierge doesn't exist, then the user will stay on the deleted task report.

Reply:
Yeah, i think that's fine. I believe it is not possible practically. Have you been able to reproduce that behavior?

Reply:
I ever reproduced that kind of behavior in the past, after login, there is only 1 report. "Clear cache and reset" from the Troubleshoot page fetches all the reports back.

Reply:
Just repro it again.

<img width="730" alt="Screenshot 2024-08-09 at 23 22 50" src="https://github.com/user-attachments/assets/612b4d8b-2fa3-4af4-93ad-3566e36d2e4a">

Reply:
Can you please describe the screenshot :) I am not getting it

Reply:
Got it. So yes, let's fallback to -1
I think we can update this line to 
https://github.com/Expensify/App/pull/46950/files#diff-ce6bb391d0654bf99a360da7d25fd73b85bb120211c8b4c2432ef7c7d8c708f3R2679
`?? -1`

Reply:
Hmm, in case the concierge chat doesn't exist, isn't it better to not navigate anywhere instead of -1 (not found)?

Reply:
Anywhere like?

Reply:
Not navigate anywhere, so the user stays at the deleted task report.

Reply:
I see, so it is the current behavior, right? If we don't return any route, user stays on the same page?

Reply:
Yes, correct

Reply:
Okay, thanks for clarifying everything 🙌 !